### PR TITLE
Remove NumPy <2 pin

### PR DESCRIPTION
**Pull request by seberg**
This PR removes the NumPy<2 pin which is expected to work for
RAPIDS projects once CuPy 13.3.0 is released (CuPy 13.2.0 had
some issues preventing the use with NumPy 2).

### CODE REVIEW

**Comment by jameslamb** [comment]
NOTE: this one shouldn't be considered done until we see a solve with NumPy `>= 2` in it.

In the latest build (run a few weeks ago), some dependencies in the environment led to `numpy==1.26.4` getting pulled in:

```text
    numpy:                            1.26.4-py311h69ead2a_0                           conda-forge   
```

([build link](https://github.com/rapidsai/integration/actions/runs/10466333613/job/28983041223?pr=718#step:7:983))

Adding `test:`-section pins of the form `numpy>=2` in the conda recipes might help expose this in CI.

File: conda/recipes/versions.yaml [hunk 1]
```diff
@@ -11,7 +11,7 @@ nccl_version:
 numba_version:
   - '>=0.57'
 numpy_version:
-  - '>=1.23,<2.0a0'
+  - '>=1.23,<3.0a0'
```
Comment (jameslamb) [inline conda/recipes/versions.yaml:14]:
```suggestion
  #- '>=1.23,<3.0a0'
  - '>=2,<3.0a0'
```

@msarahan I just updated this PR, gonna try a build with this pin to see what we can learn from the solver errors.

Comment (jameslamb) [reply conda/recipes/versions.yaml]: Most builds there failed in uninformative ways.

I've just pushed another set of commits that does the following:

* updates to latest `branch-24.10`, to get the `fmt` / `spdlog` updates from #722 
* commented out `cuxfiter` and `*cugraph*` packages

Hoping we can use that to confirm that only `cuxfilter` and `cugraph` remain for RAPIDS to be installable alongside NumPy 2.x.

Comment (jameslamb) [reply conda/recipes/versions.yaml]: 🎉 that worked!

```text
...
    numpy:                            2.0.2-py311h71ddf71_0                            conda-forge 
...
```

([build link](https://github.com/rapidsai/integration/actions/runs/11017167395/job/30594424411?pr=718#step:7:966))

I think we can now say with confidence: getting `cuxfilter` and `cugraph` supporting NumPy 2 will allow RAPIDS to support NumPy 2 😁 

(cc @jakirkham @seberg for awareness)

Comment (jameslamb) [reply conda/recipes/versions.yaml]: `cuxfilter` was merged yesterday: https://github.com/rapidsai/cuxfilter/pull/622

So just `cugraph` remains: https://github.com/rapidsai/cugraph/pull/4615

Comment (jameslamb) [reply conda/recipes/versions.yaml]: I just merged https://github.com/rapidsai/cugraph/pull/4615. Will update this again once nightly packages are built from it.

Build to watch:

https://github.com/rapidsai/cugraph/actions/runs/11107406693/job/30857924045

File: conda/recipes/versions.yaml [hunk 2]
```diff
@@ -11,7 +11,7 @@ nccl_version:
 numba_version:
   - '>=0.57'
 numpy_version:
-  - '>=1.23,<2.0a0'
+  - '>=2,<3.0a0'
```
Comment (jameslamb) [inline conda/recipes/versions.yaml:14]:
```suggestion
  - '>=1.23,<3.0a0'
```

oops, forgot to set the floor back down to `>=1.23`.

But good to see everything built successfully when it was forced to `numpy>=2`!

https://github.com/rapidsai/integration/actions/runs/11109542778/job/30865151500?pr=718

Comment (jakirkham) [reply conda/recipes/versions.yaml]: Was about to ask. Assumed this was for testing purposes

**Review by jakirkham — APPROVED** [review]
Thanks James! 🙏

**Pull request merged**